launch/launch/launch_service.py

```diff
@@ -319,10 +319,14 @@ def _on_exception(loop, context):
                     entity_futures.append(process_one_event_task)
 
                     # Wait on events and futures
-                    await asyncio.wait(
+                    completed_tasks, _ = await asyncio.wait(
                         entity_futures,
                         return_when=asyncio.FIRST_COMPLETED
                     )
+                    # Propagate exception from completed task
+                    for completed_task in completed_tasks:
+                        if completed_task.exception():
+                            raise completed_task.exception()
```

Comment:
@tumtom if more than one exception is thrown, won't we get the same "exception not retrieved" errors?

Reply:
Hi @hidmic
good question. You're probably right. Do you think this could happen and should be handled?
If so, I can think of different options:
- generate a new exception, in which we sum up the exceptions (modifies exception type!)
- log all exceptions that were retrieved, then raise the first (I think this would be better, right?)

The second approach could be implemented like this:
````python
completed_tasks_exceptions = [task.exception() for task in completed_tasks]
completed_tasks_exceptions  = list(filter(None, completed_tasks_exceptions))
if completed_tasks_exceptions:
    self.__logger.error(completed_tasks_exceptions)
    raise completed_tasks_exceptions[0]
````

Reply:
@tumtom I'm subjectively inclined towards the second approach. Perhaps we can do better when it comes to logging them, but I'm onboard with the idea.

Reply:
Ok, let me know what you think!
I'm not sure this code will ever execute due to `return_when=asyncio.FIRST_COMPLETED`, but if more than one action would complete and raise, the exceptions won't be lost now